tests/src/python/test_provider_postgres.py

```diff
@@ -1415,6 +1415,15 @@ def testCheckTidPkOnViews(self):
         for f in vl0.getFeatures():
             self.assertNotEqual(f.attribute(0), NULL)
 
+    def testFeatureCountOnView(self):
+        """
+        Test feature count on view when with estimatedmetadata = true
+        """
+        self.execSQLCommand('CREATE VIEW qgis_test.somedataview AS SELECT * FROM qgis_test."someData"')
+        vl = QgsVectorLayer(self.dbconn + ' sslmode=disable key=\'pk\' estimatedmetadata=true srid=4326 type=POINT table="qgis_test"."somedataview" (geom) sql=', 'test', 'postgres')
+        self.assertTrue(vl.isValid())
+        self.assertEqual(self.source.featureCount(), vl.dataProvider().featureCount())
```

Comment:
The test is good to have, but since  you're asking for _estimated_ data, it's a bit picky to check the estimation to be exactly equal to the actual count.

Reply:
EXPLAIN is based on statistics so if the statistics are up to date, we must have the right count, no? I force ANALYZE in my tests to be sure.

Reply:
Statistics are not exact counts. When you run ANALYZE PostgreSQL makes an estimation about the number of "live" tuples in each block of data, and applies that to the count of blocks (something like that). I guess for small tables (as in our case) it may get an exact count, but I would not bet on it :)

Reply:
Uhm, the cure is worst than the disease ! How about keeping the test but use a tolerance for estimated rows count ?

Reply:
Yes I agree but what tolerance. I was thinking about checking the returned value is just greater than 0, but can we be sure that the explain value is always greater than 0?

Reply:
I restore the test and keep only >0 has a test (if estimate fails, it returns -1)